### ISSUE TITLE
Bug: [no-non-null-assertion] wrong suggestion on left-hand side of an assignment

### ISSUE THREAD

**Issue by jendrikw** [typescript-eslint/typescript-eslint#11473]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play#ts=5.8.2&fileType=.tsx&code=CYewxgrgtgpgdgFwHQEcIwE4E8DKMA2MYCIGAFAEQCWcADhAhQJQCESAZlYQM4AEAvLzgwA7rwBiXGABkq3BGSYBuIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1s6e4XloBDZMkTR8lDk3RQx0DtEjgwAXxAqgA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
document.querySelector("input")!.files = new FileList();
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/no-non-null-assertion": "error"
  }
}a
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

When using `@typescript-eslint/no-non-null-assertion`, flagging this code is correct. But the suggested code fix is incorrect. If you apply the fix, you'll get a compiler error from tsc:

`2779: The left-hand side of an assignment expression may not be an optional property access. 1:1 - 1:39`

### Actual Result

Error, but no suggestion.

**Comment by nayounsang** [comment]
It's obvious that this is a bug, but I have a question.
Is there why tsconfig set to `"strictNullChecks": true`?
I think this option is dangerous when used with `strictNullChecks`

ref: https://typescript-eslint.io/rules/no-non-null-assertion/#when-not-to-use-it

**Comment by Josh-Cena** [comment]
? `strictNullChecks` should always be true. That section is saying that during the transition period when your types may not be correct you may have problems, which doesn't apply here.

I think we should just remove the autofix in this case.
